src/flint/fmpz_factor.jl

```diff
@@ -17,20 +17,39 @@ function _factor(a::ZZRingElem)
   return res, canonical_unit(a)
 end
 
+# Just to give a helpful error message if someone tries to factor a boolean
+function factor(b::Bool)
+  throw(DomainError("Cannot factorize a boolean"));
+end
+
+# This function handles machine integer types (up to 64 bits)
 function factor(a::T) where T <: Union{Int, UInt}
   iszero(a) && throw(ArgumentError("Argument must be non-zero"))
   u = sign(a)
   a = u < 0 ? -a : a
   F = n_factor()
   @ccall libflint.n_factor(F::Ref{n_factor}, a::UInt)::Nothing
-  res = Dict{T, Int}()
+  res = Dict{T, Int}()  # factor-multiplicity pairs
   for i in 1:F.num
     z = F.p[i]
     res[z] = F.exp[i]
   end
   return Fac(u, res)
 end
 
+# This is supposed to be called only for T in [Int128, UInt128, BigInt]
```

Comment:
I don't really understand this comment. The code clearly can also be called for e.g. `T` equal to `Int16`. So who is supposed to "only" call it for the listed types? What happens if I call it with another Julia `Integer` type? Is there a problem (then we should address it)? Or is actually all fine (then remove this comment?)